authentication/authentication.py

```diff
@@ -0,0 +1,32 @@
+from google.oauth2 import id_token
+from google.auth.transport import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework import authentication, exceptions
+import os 
+import requests
+User = get_user_model()
+
+class GoogleIDTokenAuthentication(authentication.BaseAuthentication):
+    def authenticate(self, request):
+        access_token = request.COOKIES.get('access_token')
+        if not access_token:
+            raise exceptions.AuthenticationFailed('No access token in cookies')
+        try:
+            response = requests.get(f'https://www.googleapis.com/oauth2/v1/tokeninfo?access_token={access_token}')
+            if response.status_code != 200:
+                raise exceptions.AuthenticationFailed('Invalid token')
```

Comment:
if token is invalid doesn't mean user is not authenticated. you need to also check if refresh_token is valid or not. if refresh_token is valid, then create a new access_token for user and continue with request. if refresh_token is also invalid then return a not authenticated error